setup.py

```diff
@@ -1,91 +1,81 @@
-from setuptools import setup, extension
-from setuptools.command.build_ext import build_ext
-import sys
+# Copyright 2024 D-Wave Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
 import os
 import platform
-cwd = os.path.abspath(os.path.dirname(__file__))
+import sys
 
-if not os.path.exists(os.path.join(cwd, 'PKG-INFO')):
-    try:
-        from Cython.Build import cythonize
-        USE_CYTHON = True
-    except ImportError:
-        USE_CYTHON = False
-else:
-    USE_CYTHON = False
+from Cython.Build import cythonize
+from setuptools import setup, Extension
 
-_PY2 = sys.version_info.major == 2
 
-# Change directories so this works when called from other locations. Useful in build systems that build from source.
+# Change directories so this works when called from other locations.
+# Useful in build systems that build from source.
 setup_folder_loc = os.path.dirname(os.path.abspath(__file__))
 os.chdir(setup_folder_loc)
 
-# Add __version__, __author__, __authoremail__, __description__ to this namespace
-path_to_package_info = os.path.join('.', 'minorminer', 'package_info.py')
-if _PY2:
-    execfile(path_to_package_info)
-else:
-    exec(open(path_to_package_info).read())
-
-base_compile_args = {
-    'msvc': ['/std:c++latest', '/MT', '/EHsc', '/O2'],
-    'unix': ['-std=c++17', '-Wall', '-Wno-format-security', '-Ofast', '-fomit-frame-pointer',
-             '-DNDEBUG'],
```

Comment:
Note `-DNDEBUG` is included by default by `setuptools`, so I didn't keep it. I also didn't add `-UNDEBUG` to debug mode, in order to keep the current functionality. Though we probably should in a followup, preferably one that includes debug mode testing in CI.